drivers/bridge/bridge.go

```diff
@@ -46,6 +42,8 @@ type Configuration struct {
 	EnableICC             bool
 	EnableIPForwarding    bool
 	AllowNonDefaultBridge bool
+	// Driver components
+	PortAllocator *portallocator.PortAllocator
```

Comment:
I have couple of points towards this
1. To me, maintaining this database itself seems redundant and can easily be incorrect, given the fact that the actual source of truth is the Operating System actually allocates, reserves and manages the port(s). In addition, the Port Allocator doesnt allocate anything. Rather it just lets the application know "its" view of port availability in its local db and not querying the OS.
2. With the above point, if we dont make it a singleton for that particular namespace, then we are essentially not doing the right thing. Also, the argument on `could be used on the default as well as on other net namespaces` can be achieved in different ways and I dont see it as a valid reason for not making it a singleton.